src/components/Container.js

```diff
@@ -114,25 +113,45 @@ class Container extends React.Component{
         const coords = Projection.toLonLat(_.get(drawEvent, 'target.sketchCoords_'));
         const newHydrants = _.clone(hydrants);
         const createdHydrant = {
-          id: new Date().getTime(), 
-          coords: coords, 
-          trail: selectedTrail
+          id: new Date().getTime(),
+          coords: coords,
+          trail: selectedTrail,
+          elevation: null,
+          name: null,
         };
         newHydrants[createdHydrant.id] = createdHydrant;
         this.setState({hydrants: newHydrants});
-        getElevation(coords).then((data) => {
-          const elevation = data[0].height;
-          const newHydrantsv2 = _.clone(newHydrants);
-          newHydrantsv2[createdHydrant.id].elevation = elevation;
-          this.setState({hydrants: newHydrantsv2});
-        });
+        getElevation(coords)
+          .then((data) => {
+              const elevation = data[0].height;
+              const newHydrantsv2 = _.clone(newHydrants);
+              newHydrantsv2[createdHydrant.id].elevation = elevation;
+              this.setState({hydrants: newHydrantsv2});
+            })
+            .then(this.indexNamebyElevation);
         break;
       }
       default:
         console.log("haven't implemented this type yet");
     }
   }
 
+  indexNamebyElevation = () => {
+    const { selectedTrail } = this.state;
+    let newHydrants = _.cloneDeep(this.state.hydrants)
+    const targetHydrants = _.pickBy(newHydrants, (h) => h['trail'] === selectedTrail)
+    const sortedElevHydrants = _.orderBy(targetHydrants,'elevation', 'desc').map((h,i)=> {
+      h.name = i + 1
+      return h
+    })
+    _.each(sortedElevHydrants, (h,key)=> {
+      newHydrants[h.id] = h
+    })
```

Comment:
Try chaining them!
```
newHydrants = _(hydrants)
  .pickBy(() => {})
  .orderBy(() => {})
  .map(() => {})
```

Reply:
hot damn, that looks nice. Figured there was a way of chaining but wasn't sure what the right lodash syntax was.